lang/java/reef-bridge-java/src/main/java/org/apache/reef/javabridge/generic/JobDriver.java

```diff
@@ -147,13 +149,15 @@
             final DriverStatusManager driverStatusManager,
             final LoggingScopeFactory loggingScopeFactory,
             final LibLoader libLoader,
-            final LocalAddressProvider localAddressProvider) {
+            final LocalAddressProvider localAddressProvider,
+            final ActiveContextBridgeFactory activeContextBridgeFactory) {
```

Comment:
Agreed. I think in the long term, having a factory for all these wrappers would be good. That way, we can manage their memory use very carefully. I actually went down that path in my 1st attempt at this. It just turned into an all-out overhaul of the bridge, which is something I'd rather avoid in a single PR. We could go down that path one step at a time, though. Once this is in, we can turn `ActiveContextBridgeFactory` into `ContextBridgeFactory` and add the other context wrappers. After that, we add tasks etc.